src/ir/call_resolver.cpp

```
@@ -0,0 +1,211 @@
+                        .unsupported_reason = std::move(reason),
+                        .contract = {}};
+}
+
+ResolvedCall resolve_helper(const Call& call, const ProgramInfo& info) {
+    // Matches the unmarshal-time branching: a helper whose id is known to the
+    // platform but unusable for *this* program type must surface as "helper
+    // function is unavailable on this platform", not the generic "prototype"
+    // diagnostic. Critical for program-type-specific gates (e.g.,
+    // bpf_get_socket_cookie is allowed from cgroup/connect4 but not xdp).
+    if (!info.platform->is_helper_usable(call.func, info.type)) {
+        std::string name = std::to_string(call.func);
```

Clang (14.0.6)

[warning] variable 'name' is not initialized

(cppcoreguidelines-init-variables)